packages/sdk-core/test/5_subgraph.test.ts

```diff
@@ -36,5 +37,35 @@ describe("Subgraph Tests", () => {
         it("Should be able to use the listener", async () => {
             await testListenerInitialization(query);
         });
+
+        it("Should have the correct subgraph endpoints", async () => {
+            const resolverDataArray = Array.from(
+                chainIdToResolverDataMap.values()
+            );
+            await Promise.all(
+                resolverDataArray.map(async (x) => {
+                    // @note this handles arbitrum-goerli not being ready
+                    if (!x.subgraphAPIEndpoint.includes("arbitrum-goerli")) {
+                        if (
+                            // @note this handles feature endpoint only having goerli/matic endpoints
+                            (process.env.SUBGRAPH_RELEASE_TAG === "feature" &&
+                                (x.networkName === "eth-goerli" ||
+                                    x.networkName === "polygon-mainnet")) ||
+                            process.env.SUBGRAPH_RELEASE_TAG !== "feature"
+                        ) {
+                            const query = new Query({
+                                customSubgraphQueriesEndpoint:
+                                    x.subgraphAPIEndpoint,
+                            });
+                            const event = await query.listEvents(
+                                {},
+                                { take: 1 }
+                            );
+                            expect(event.data.length).to.be.greaterThan(0);
+                        }
```

Comment:
The deep nesting makes the test a bit complicated. I can imagine it's easily possible that the test will always pass because it never gets through all the conditionals. Maybe it can be simplified a bit?